### ISSUE TITLE
Client certificate support partially breaks after updating to System.Net.Http 4.3.1

### ISSUE THREAD

**Issue by harrdou** [aspnet/IISIntegration#335]
I ran into  [this](https://github.com/dotnet/corefx/issues/11100) issue with Microsoft.Extensions.Configuration.AzureKeyVault the after upgrading to Visual Studio 2017. Upgrading my web application  to use System.Net.Http 4.3.1 solved that problem, but it introduced a new one with the IIS Integration.

With System.Net.Http 4.3.1, I am seeing the client certificate forwarded correctly with some requests, but not all. Specifically, the client certificate is passed as null for any requests for static "*.js.map" files.

**Comment by Tratcher** [comment]
Are you sure those files are requested over https?

**Comment by harrdou** [comment]
Yes, but I've also confirmed that the client certificate is actually missing from the request; so it's not an issue with the IISIntegration middleware. It seems these requests are originating from the Visual Studio debugger as part of its integration with IE for debugging JavaScript. It looks like the newer tooling doesn't consistently use the same TLS session as the actual application pages: sometimes the client certificate is included, sometimes it is not.

I don't use the JavaScript debugging feature so I just turned it off and I'm back in business.
